From: Chad Monroe <chad@monroe.io>
Date: Fri, 11 Oct 2024 16:32:55 -0700
Subject: [PATCH] net: dsa: mxl862xx: add some hacks for testing

1. add BR_PORT_LOCKED flag but do nothing with it
2. fix compile warning for unused dsa_8021q_rx_vid

Signed-off-by: Chad Monroe <chad@monroe.io>
---
 drivers/net/dsa/mxl862xx/mxl862xx.c |    8 ++++----
 net/dsa/tag_mxl862xx_8021q.c        |    2 ++
 2 files changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/net/dsa/mxl862xx/mxl862xx.c
+++ b/drivers/net/dsa/mxl862xx/mxl862xx.c
@@ -4294,10 +4294,10 @@ static int mxl862xx_port_pre_bridge_flag
 {
 	int ret = 0;
 
-	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
+	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
 		dev_err(ds->dev, "%s: Port:%d unsupported bridge flags:0x%lx\n",
 				__func__, port, flags.mask);
-		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
+		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
 			ret = -EINVAL;
 			NL_SET_ERR_MSG_MOD(extack, "Unsupported bridge flags:0x%lx");
 		}
@@ -4320,10 +4320,10 @@ static int mxl862xx_port_bridge_flags(st
 
 	/* .port_pre_bridge_flags is called after this function,
 	 *  so the supported flags check is needed also here */
-	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
+	if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
 		dev_err(ds->dev, "%s: Port:%d unsupported bridge flags:0x%lx\n",
 				__func__, port, flags.mask);
-		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING)) {
+		if (flags.mask & ~(BR_FLOOD | BR_MCAST_FLOOD | BR_BCAST_FLOOD | BR_LEARNING | BR_PORT_LOCKED)) {
 			ret = -EINVAL;
 			goto EXIT;
 		}
--- a/net/dsa/tag_mxl862xx_8021q.c
+++ b/net/dsa/tag_mxl862xx_8021q.c
@@ -58,10 +58,12 @@
  * DSA tx/rx vid functions are not avaliable, so dummy
  * functions are here to make the code compilable. */
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION (5, 16, 0))
+#if 0
 static u16 dsa_8021q_rx_vid(struct dsa_switch *ds, int port)
 {
    return 0;
 }
+#endif
 
 static u16 dsa_8021q_tx_vid(struct dsa_switch *ds, int port)
 {
